src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java

```diff
@@ -33,19 +33,30 @@
   @Autowired private CaseService caseService;
 
   /**
-   * the GET end point to return the latest valid Non HI Case by UPRN
+   * the GET end point to return latest valid Case which matches the supplied sample attribute
+   * name/value.
    *
-   * @param uprn the UPRN
+   * @param attributeKey - is the name of the field in the sample data to search by.
+   * @param attributeValue - is the value that target case(s) must contain.
    * @return Returned latest Non HI case with valid address for the UPRN
    * @throws CTPException something went wrong - thrown by case service
    */
-  @RequestMapping(value = "/uprn/{uprn}", method = RequestMethod.GET)
-  public ResponseEntity<CaseDTO> getCaseByUPRN(
-      @PathVariable(value = "uprn") final UniquePropertyReferenceNumber uprn) throws CTPException {
-    log.info("Entering GET getLatestValidNonHICaseByUPRN", kv("pathParam.uprn", uprn));
-
-    CaseDTO result = caseService.getLatestValidCaseByUPRN(uprn);
-    log.debug("Exit GET getLatestValidNonHICaseByUPRN", kv("pathParam.uprn", uprn));
+  @RequestMapping(value = "attribute/{attributeKey}/{attributeValue}", method = RequestMethod.GET)
+  public ResponseEntity<List<CaseDTO>> getCaseByAttribute(
```

Comment:
the method name suggests just getting one case, but we are returning a list of cases.  Suggest `listCasesByAttribute` or `findCasesByAttribute`

Reply:
Renamed to findCasesByAttribute()